backend/bitsurance/bitsurance.go

```diff
@@ -35,38 +37,41 @@ const (
 	widgetTestURL = "https://test.bitsurance.eu/?wallet=bitbox&version=" + widgetVersion + "&lang="
 )
 
-const statusActive = "active"
+type DetailStatus string
 
-type accountDetails struct {
-	Status string `json:"status"`
+const (
+	ActiveStatus      DetailStatus = "active"
+	ProcessingStatus  DetailStatus = "processing"
+	RefusedStatus     DetailStatus = "refused"
+	WaitPaymentStatus DetailStatus = "waitpayment"
+	InactiveStatus    DetailStatus = "inactive"
+	CanceledStatus    DetailStatus = "canceled"
+)
+
+type AccountDetails struct {
+	AccountCode types.Code   `json:"code"`
+	Status      DetailStatus `json:"status"`
+	Details     struct {
+		MaxCoverage          int    `json:"maxCoverage"`
+		MaxCoverageFormatted string `json:"maxCoverageFormatted"`
+		Currency             string `json:"currency"`
+		Support              string `json:"support"`
+	} `json:"details"`
 }
 
-// bitsuranceCheckId fetches the account details of the passed accountId from the
-// Bitsurance server. It returns true if the account status is "active". Possible
-// status are (from Bitsurance Api docs):
-// - "processing"	when still in creation/checking phase (short time)
-// - "refused" application got refused
-// - "waitpayment" accepted, but waiting on first payment
-// - "active" insurance coverage activated
-// - "inactive" offer support link for more info (maybe missed payment)
-// - "canceled" contract got canceled
-//
+// bitsuranceCheckId fetches and returns the account details of the passed accountId from the
+// Bitsurance server.
 // If an accountId is not retrieved at all, the endpoint return a 404 http code.
-func bitsuranceCheckId(httpClient *http.Client, accountId string) (bool, error) {
+func bitsuranceCheckId(httpClient *http.Client, accountId string) (AccountDetails, error) {
 	endpoint := apiURL + "accountDetails/" + accountId
-	details := accountDetails{}
-	code, err := util.APIGet(httpClient, endpoint, apiKey, 1024, &details)
+	account := AccountDetails{}
+	code, err := util.APIGet(httpClient, endpoint, apiKey, 1024, &account)
 	if err != nil && code != http.StatusNotFound {
-		return false, err
+		return account, err
 	}
-	if code == http.StatusNotFound {
-		return false, nil
-	}
-
-	if details.Status == statusActive {
-		return true, nil
-	}
-	return false, nil
+	ratAmount := new(big.Rat).SetInt64(int64(account.Details.MaxCoverage))
+	account.Details.MaxCoverageFormatted = coin.FormatAsCurrency(ratAmount, account.Details.Currency, false)
```

Comment:
seems like the currency field for now is unused except if it is BTC? maybe it is safer to convert the param to `isBTC bool` then, otherwise things might silently break if we start using the currency parameter for other things (e.g. we don't know exactly if the bitsurance Currency field maps perfectly to our own list of fiat currencies).

Reply:
Done, I made a dedicated commit and took the chance to refactor a bit, renaming a few confusing variables.